### ISSUE TITLE
Migrate to Mantle TestKit

### ISSUE THREAD

**Issue by attackant** [alleyinteractive/apple-news#1022]
### Description

Apple News currently uses an older test framework. We migrate it to use mantle-framework/testkit which is our standard modern test framework.

## Tasks

- [x] Audit current test codebase
- [x] Migrate test code to mantle-framework/testkit
- [x] Update build scripts and dependencies 
- [x] Validate new tests pass
- [ ] Remove old test framework code

## Requirements

- Apple News tests should have feature parity with old test framework
- All existing tests should pass after migration
- No disruption to developers using test framework
- Easy to maintain and extend tests in new framework

### Use Case

The mantle-framework/testkit is the standard test framework used by Alley projects. By migrating Apple News to use this framework, we align and standardize testing approaches across all Alley products.

Some benefits of standardizing on the mantle-framework/testkit:

- Developers can more easily work across different codebases because testing approaches are consistent.
- Knowledge, tools, and best practices are shared across all applications.
- New hires are only required to learn one test framework.
- Code is interoperable between applications, allowing for easier migration of tests.
- Consolidates maintenance and reduces overhead of supporting multiple frameworks.
- Brings Apple News up to the current high-quality testing bar set by other Alley apps.

In summary, migrating Apple News to the standard testkit benefits Alley by aligning and improving testing practices company-wide rather than having disparate approaches. It enforces uniform processes for writing, running, and maintaining tests.

**Comment by attackant** [comment]
I briefly looked into this for the 2.4.1 release but think it would be too time consuming, but thought it would be worth adding to our backlog.
